ocaml/xapi/vm_platform.ml

```diff
@@ -198,19 +198,16 @@ let sanity_check ~platformdata ?firmware ~vcpu_max ~vcpu_at_startup ~domain_type
             (Api_errors.Server_error
                ( Api_errors.invalid_value
                , [
-                   Printf.sprintf "platform:cores-per-socket (value %d)" cps
-                 ; Printf.sprintf
-                     "VCPUs_max (value %d) must be a multiple of \
-                      cores-per-socket"
-                     vcpus
+                   Printf.sprintf "platform:cores-per-socket"
```

Comment:
Might be useful to still log the more detailed reason why the failure occured.
(Or introduce a new kind of API error that can give more details about why a value is wrong)

Reply:
There's a similar case with certificates and validity dates:
There are 2 API errors, `server_certificate_not_valid_yet` and `server_certificate_expired` each one has two parameters. The erroneous the refused value, and the the closest valid value. These can be used to construct message that can be easily understood.
You can see the errors defined in
https://github.com/xapi-project/xen-api/blob/99d98bdd3c00e7f5053940d3d3430289c01ecee3/ocaml/idl/datamodel_errors.ml#L1051-L1054

Reply:
Thank you so much for that. :slightly_smiling_face: